### PR TITLE
Use whitelist from configuration to allow EU Access

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**
We allow EU Access for selected  and whitelisted GlobalAccountIds. If GlobalAccountId is not whitelisted we return Http Status 400 and informative message.

Changes proposed in this pull request:

- Whitelist definition in values.yaml
- Whitelist usage in app-config.yaml and deployment
- New package eu_access with helper function for whitelist reading

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/eu_access/whitelisted_globalaccounts.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func IsNotWhitelisted(globalAccountId string, whitelist WhitelistSet) bool {
+	_, found := whitelist[globalAccountId]
+	return !found
+}
+
+func ReadWhitelistedGlobalAccountIdsFromFile(filename string) (WhitelistSet, error) {
+	var whitelist, err = os.ReadFile(filename)
+	yamlData := make(map[string][]string)
+	whitelistSet := WhitelistSet{}
+	if err != nil {
```
Comment (ralikio) [inline components/kyma-environment-broker/internal/eu_access/whitelisted_globalaccounts.go:25]: Can we move this err higher up?

File: components/kyma-environment-broker/internal/eu_access/whitelisted_globalaccounts_test.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadWhitelistedGlobalAccountIdsFromFile(t *testing.T) {
+	// given/when
+	d, err := ReadWhitelistedGlobalAccountIdsFromFile("test/eu_access_whitelist.yaml")
```
Comment (ralikio) [inline components/kyma-environment-broker/internal/eu_access/whitelisted_globalaccounts_test.go:12]: Can we add case where we test multiple entries of whitelisted accounts?

File: components/kyma-environment-broker/internal/broker/instance_create.go [hunk 3]
```diff
@@ -76,20 +83,22 @@ func NewProvision(cfg Config,
 	}
 
 	return &ProvisionEndpoint{
```
Comment (ralikio) [inline components/kyma-environment-broker/internal/broker/instance_create.go:85]: Does it make sensible to apply new validation to other operations as well?

**Review by ralikio — CHANGES_REQUESTED** [review]

File: components/kyma-environment-broker/cmd/broker/main.go [hunk 4]
```diff
@@ -14,6 +14,8 @@ import (
 	"sort"
 	"time"
 
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/eu_access"
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/cmd/broker/main.go:17]: Renamed to `euaccess`.

File: components/kyma-environment-broker/internal/eu_access/whitelisted_globalaccounts_test.go (hunk 2 above)
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/eu_access/whitelisted_globalaccounts_test.go:12]: Done.

File: components/kyma-environment-broker/internal/broker/instance_create.go (hunk 3 above)
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create.go:85]: Don't get it. We need to talk.

Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create.go:85]: Discussed in private communication.

**Review by ralikio — APPROVED** [review]

**Pull request merged**